### PR TITLE
Remove deprecated fiftyone-desktop refs

### PR DESCRIPTION
**Pull request by brimoor**
Resolves https://github.com/voxel51/fiftyone/issues/4880
Resolves https://github.com/voxel51/fiftyone/issues/4889

## Tested by

```
fiftyone quickstart
```

## Summary by CodeRabbit

- **New Features**
	- Simplified the quickstart process by removing the desktop application option from the interface.
- **Bug Fixes**
	- Eliminated fallback logic for the `FIFTYONE_DESKTOP_APP_DIR`, streamlining the app setup process.
- **Improvements**
	- Enhanced session management by standardizing the waiting behavior across different session types.

### CODE REVIEW

File: fiftyone/constants.py [hunk 1]
```diff
@@ -106,11 +106,3 @@
 
 # Server setup
 SERVER_DIR = os.path.join(FIFTYONE_DIR, "server")
-
```
Comment (brimoor) [inline fiftyone/constants.py:109]: @benjaminpkane just confirming that this is safe to remove: it doesn't appear to be used

**Review by brimoor — COMMENTED** [review]
@benjaminpkane when I run `fiftyone quickstart` on this branch, the App now launches but it appears to immediately exit. Can you reproduce? Is `session.wait()` also broken?

File: fiftyone/constants.py (hunk 1 above)
Comment (benjaminpkane) [reply fiftyone/constants.py:109]: Yes, no used

**Review by benjaminpkane — APPROVED** [review]

**Comment by benjaminpkane** [comment]
> @benjaminpkane when I run `fiftyone quickstart` on this branch, the App now launches but it appears to immediately exit. Can you reproduce? Is `session.wait()` also broken?

Fixed. My mistake

**Review by benjaminpkane — APPROVED** [review]

**Pull request merged**